tests/entity/test_order.py

```diff
@@ -343,3 +343,846 @@ def test_order_currency_stripped(self):
         # self.assertEqual(order.refund_total, 5555.99)
         # self.assertEqual(order.subscription_discount, 6666.99)
         self.assertEqual(order.grand_total, 7777.99)
+
+    def test_order_promotion_applied(self):
+        # GIVEN
+        html = """
+<div id="orderDetails" class="a-section dynamic-width">
```

Comment:
Can you move this in to its own HTML file and read it, like other tests? Then this looks great, and happy to merge the PR!

Reply:
Yup. Do you want me to refactor the other pre-existing test in `test_order.py` while I'm in there?

Reply:
That would be awesome, thank you!

Reply:
This isn't a blocker, so going to merge this, as I want to include this addition in a release for another bug fix. If you've made these changes for this or other tests, feel free to push another PR.